src/RuleMatcher.cc

```diff
@@ -842,7 +843,12 @@ void RuleMatcher::Match(RuleEndpointState* state, Rule::PatternType type, const
     // matched patterns per connection (which is a plausible assumption).
 
     // Find rules for which patterns have matched.
-    set<pair<Rule*, MatchPos>> rule_matches;
+    auto cmp = [](pair<Rule*, MatchPos> a, pair<Rule*, MatchPos> b) {
+        if ( *a.first == *b.first )
+            return a.second < b.second;
+        return *a.first < *b.first;
+    };
+    set<pair<Rule*, MatchPos>, decltype(cmp)> rule_matches(cmp);
```

Comment:
Thanks for looking and fixing. Rule's apparently have a unique `Index()` which should be consistent across platforms (haven't tried).

That would allow to do the following instead:
```
     // Find rules for which patterns have matched.
-    set<pair<Rule*, MatchPos>> rule_matches;
+    map<decltype(Rule::idx), std::pair<Rule*, MatchPos>> rule_matches;
```

I doubt this matters performance wise, but this would sort by Index() which is only a 4 byte unsigned int.

Reply:
I saw those and was suspicious about the notion of an index for ordering, but it might be fine. I can take a look.

Reply:
I took a look and the rule indexes are determined the parser, numbering them up as it encounters them. That seems safe, so I switched it over. CI agreed in my fork, fingers crossed. Thanks Arne.